### ISSUE TITLE
clinical attribute type handling

### ISSUE THREAD

**Issue by n1zea144** [cBioPortal/cbioportal#5559]
I was recently asked by GENIE folks if the cBioPortal could support "Unknown" and "Not Collected" values across all clinical attributes.  In parallel I was tracking down an issue with an incorrectly computed "Patient Current Age" and "Age at Which Sequencing was Reported" for a PEDs case (in PEDs cases we typically display ages as "<18" although we are going to lift this restriction soon - but thats another story).

As I addressed these issues, I had to reach out to both @alisman and @zhx828 to determine what support, if any, the patient view and study view had for numerical data that contain a mixture of  number/string, "<18" or string entirely, "Not Collected".

As most of you know, the patient view treats all clinical data as strings (ignoring the clinical metadata datatype property) and the study view uses the datatype to choose a pie chart on bar chart for display purposes.  It turns out that the latest study view has no restriction on values contained in a numerical field.  For example, it nows supports a birth year with strings that represent when a birth year is not provided (thanks for pointing this out Hongxin!):

![image](https://user-images.githubusercontent.com/366003/50706054-b3a33800-102a-11e9-836e-15e672aac40a.png)

and because the patient view treats values as string, we can have values like "<18" as an age:

![image](https://user-images.githubusercontent.com/366003/50706152-054bc280-102b-11e9-9d31-3dca34984f7e.png)

In addition, we recently introduced a clinical attribute which represents a URL to a path slide viewer, but the URL is not clickable in places the attribute is displayed (of course whether or not the URL should be displayed in all places is a question):

![image](https://user-images.githubusercontent.com/366003/50706581-409ac100-102c-11e9-9b3a-60262daca930.png)

As conversations unfolded, it occurred to me that there isn't a uniform approach or adherence to the datatype of a clinical attribute (this may not be possible as the previous examples show) and in other cases maybe a new datatype should be introduced.  

My feeling is as we collect more clinical annotation and longitudinal treatment information, a more systematic approach to type handling of clinical attributes across the site will be important.  Am I creating a mountain out of a molehill or do others feel this way?
